Python Project 
Car Data Analysis 


In [4]:
import pandas as pd

In [40]:
cars= pd.read_csv("../HP/Downloads/Project+2+-+Cars+Dataset.csv")

OPEN DATA SET FOR CAR ANALYSIS

In [42]:
cars

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
427  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
428  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
429  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
430  Volvo                      V40  Wagon  Europe      Front  $26,135    
431  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0       265.0      17.0         23.0   
1    $21,761          2.0        4.0       200.0      24.0         31.0   
2    $24,647          2.4        4.0       200.0      22.0         29.0   
3    $30,299          3.2        6.0       270.0      20.0         28.0   
4    $39,014          3.5        6.0       225.0      18.0         24.0   
..        ...         ...        ...         ...       ...          ...   
427  $38,203          2.4        5.0       197.0      21.0         28.0   
428  $40,083          2.3        5.0       242.0      20.0         26.0   
429  $42,573          2.9        6.0       268.0      19.0         26.0   
430  $24,641          1.9        4.0       170.0      22.0         29.0   
431  $33,112          2.5        5.0       208.0      20.0         27.0   

     Weight  Wheelbase  Length  
0    4451.0      106.0   189.0  
1    2778.0      101.0   172.0  
2    3230.0      105.0   183.0  
3    3575.0      108.0   186.0  
4    3880.0      115.0   197.0  
..      ...        ...     ...  
427  3450.0      105.0   186.0  
428  3450.0      105.0   186.0  
429  3653.0      110.0   190.0  
430  2822.0      101.0   180.0  
431  3823.0      109.0   186.0  

[432 rows x 15 columns]

TO CHECK ROWS AND COLUMN OF DATASET

THERE ARE 432 rows and 15 Column

In [44]:
cars.shape

(432, 15)

CHECK FOR NULL VALUES AND THEN REPLACE NULL WITH MEAN OF DATA 

In [46]:
cars.isnull()

Make  Model   Type  Origin  DriveTrain   MSRP  Invoice  EngineSize  \
0    False  False  False   False       False  False    False       False   
1    False  False  False   False       False  False    False       False   
2    False  False  False   False       False  False    False       False   
3    False  False  False   False       False  False    False       False   
4    False  False  False   False       False  False    False       False   
..     ...    ...    ...     ...         ...    ...      ...         ...   
427  False  False  False   False       False  False    False       False   
428  False  False  False   False       False  False    False       False   
429  False  False  False   False       False  False    False       False   
430  False  False  False   False       False  False    False       False   
431  False  False  False   False       False  False    False       False   

     Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  Wheelbase  Length  
0        False       False     False        False   False      False   False  
1        False       False     False        False   False      False   False  
2        False       False     False        False   False      False   False  
3        False       False     False        False   False      False   False  
4        False       False     False        False   False      False   False  
..         ...         ...       ...          ...     ...        ...     ...  
427      False       False     False        False   False      False   False  
428      False       False     False        False   False      False   False  
429      False       False     False        False   False      False   False  
430      False       False     False        False   False      False   False  
431      False       False     False        False   False      False   False  

[432 rows x 15 columns]

In [59]:
car.isnull().sum()

Make           4
Model          4
Type           4
Origin         4
DriveTrain     4
MSRP           4
Invoice        4
EngineSize     4
Cylinders      0
Horsepower     4
MPG_City       4
MPG_Highway    4
Weight         4
Wheelbase      4
Length         4
dtype: int64

In [48]:
cars['Cylinders'].fillna(car['Cylinders'].mean(), inplace=True)

C:\Users\HP\AppData\Local\Temp\ipykernel_3716\1372909840.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  cars['Cylinders'].fillna(car['Cylinders'].mean(), inplace=True)


TO CHECK OCCURENCE OF BRAND (MAKE) OF CARS IN THE DATASET

In [62]:
cars['Make'].value_counts()

Make
Toyota           28
Chevrolet        27
Mercedes-Benz    26
Ford             23
BMW              20
Audi             19
Honda            17
Nissan           17
Volkswagen       15
Chrysler         15
Dodge            13
Mitsubishi       13
Volvo            12
Jaguar           12
Hyundai          12
Subaru           11
Pontiac          11
Mazda            11
Lexus            11
Kia              11
Buick             9
Mercury           9
Lincoln           9
Saturn            8
Cadillac          8
Suzuki            8
Infiniti          8
GMC               8
Acura             7
Porsche           7
Saab              7
Land Rover        3
Oldsmobile        3
Jeep              3
Scion             2
Isuzu             2
MINI              2
Hummer            1
Name: count, dtype: int64

SHOW ALL RECORDS WHERE ORIGIN IS USA OR EUROPE, THEN DO A SEPERATE ONE FOR ASIA AND EUROPE

In [77]:
cars[cars['Origin'].isin(['USA', 'Europe'])]

Make                      Model   Type  Origin DriveTrain      MSRP  \
7     Audi                A4 1.8T 4dr  Sedan  Europe      Front  $25,940    
8     Audi     A41.8T convertible 2dr  Sedan  Europe      Front  $35,940    
9     Audi                 A4 3.0 4dr  Sedan  Europe      Front  $31,840    
10    Audi  A4 3.0 Quattro 4dr manual  Sedan  Europe        All  $33,430    
11    Audi    A4 3.0 Quattro 4dr auto  Sedan  Europe        All  $34,480    
..     ...                        ...    ...     ...        ...       ...   
427  Volvo    C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
428  Volvo    C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
429  Volvo                 S80 T6 4dr  Sedan  Europe      Front  $45,210    
430  Volvo                        V40  Wagon  Europe      Front  $26,135    
431  Volvo                       XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
7    $23,508          1.8        4.0       170.0      22.0         31.0   
8    $32,506          1.8        4.0       170.0      23.0         30.0   
9    $28,846          3.0        6.0       220.0      20.0         28.0   
10   $30,366          3.0        6.0       220.0      17.0         26.0   
11   $31,388          3.0        6.0       220.0      18.0         25.0   
..        ...         ...        ...         ...       ...          ...   
427  $38,203          2.4        5.0       197.0      21.0         28.0   
428  $40,083          2.3        5.0       242.0      20.0         26.0   
429  $42,573          2.9        6.0       268.0      19.0         26.0   
430  $24,641          1.9        4.0       170.0      22.0         29.0   
431  $33,112          2.5        5.0       208.0      20.0         27.0   

     Weight  Wheelbase  Length  
7    3252.0      104.0   179.0  
8    3638.0      105.0   180.0  
9    3462.0      104.0   179.0  
10   3583.0      104.0   179.0  
11   3627.0      104.0   179.0  
..      ...        ...     ...  
427  3450.0      105.0   186.0  
428  3450.0      105.0   186.0  
429  3653.0      110.0   190.0  
430  2822.0      101.0   180.0  
431  3823.0      109.0   186.0  

[270 rows x 15 columns]

In [75]:
cars[cars['Origin'].isin(['Asia', 'Europe'])]

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
427  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
428  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
429  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
430  Volvo                      V40  Wagon  Europe      Front  $26,135    
431  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0       265.0      17.0         23.0   
1    $21,761          2.0        4.0       200.0      24.0         31.0   
2    $24,647          2.4        4.0       200.0      22.0         29.0   
3    $30,299          3.2        6.0       270.0      20.0         28.0   
4    $39,014          3.5        6.0       225.0      18.0         24.0   
..        ...         ...        ...         ...       ...          ...   
427  $38,203          2.4        5.0       197.0      21.0         28.0   
428  $40,083          2.3        5.0       242.0      20.0         26.0   
429  $42,573          2.9        6.0       268.0      19.0         26.0   
430  $24,641          1.9        4.0       170.0      22.0         29.0   
431  $33,112          2.5        5.0       208.0      20.0         27.0   

     Weight  Wheelbase  Length  
0    4451.0      106.0   189.0  
1    2778.0      101.0   172.0  
2    3230.0      105.0   183.0  
3    3575.0      108.0   186.0  
4    3880.0      115.0   197.0  
..      ...        ...     ...  
427  3450.0      105.0   186.0  
428  3450.0      105.0   186.0  
429  3653.0      110.0   190.0  
430  2822.0      101.0   180.0  
431  3823.0      109.0   186.0  

[281 rows x 15 columns]

In [71]:
cars['Origin'].value_counts()

Origin
Asia      158
USA       147
Europe    123
Name: count, dtype: int64

TO REMOVE ALL THE RECORDS WITH WEIGHT ABOVE 4000

In [80]:
cars[~(cars['Weight'] > 4000)]

Make                    Model   Type  Origin DriveTrain      MSRP  \
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
5    Acura  3.5 RL w/Navigation 4dr  Sedan    Asia      Front  $46,100    
..     ...                      ...    ...     ...        ...       ...   
427  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
428  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
429  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
430  Volvo                      V40  Wagon  Europe      Front  $26,135    
431  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
1    $21,761          2.0        4.0       200.0      24.0         31.0   
2    $24,647          2.4        4.0       200.0      22.0         29.0   
3    $30,299          3.2        6.0       270.0      20.0         28.0   
4    $39,014          3.5        6.0       225.0      18.0         24.0   
5    $41,100          3.5        6.0       225.0      18.0         24.0   
..        ...         ...        ...         ...       ...          ...   
427  $38,203          2.4        5.0       197.0      21.0         28.0   
428  $40,083          2.3        5.0       242.0      20.0         26.0   
429  $42,573          2.9        6.0       268.0      19.0         26.0   
430  $24,641          1.9        4.0       170.0      22.0         29.0   
431  $33,112          2.5        5.0       208.0      20.0         27.0   

     Weight  Wheelbase  Length  
1    2778.0      101.0   172.0  
2    3230.0      105.0   183.0  
3    3575.0      108.0   186.0  
4    3880.0      115.0   197.0  
5    3893.0      115.0   197.0  
..      ...        ...     ...  
427  3450.0      105.0   186.0  
428  3450.0      105.0   186.0  
429  3653.0      110.0   190.0  
430  2822.0      101.0   180.0  
431  3823.0      109.0   186.0  

[329 rows x 15 columns]

TO INCREASE THE VALUE OF MPG_CITY COLUMN BY 3

In [83]:
cars['MPG_City'] =  cars['MPG_City'].apply(lambda x:x+3)

In [85]:
cars

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
427  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
428  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
429  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
430  Volvo                      V40  Wagon  Europe      Front  $26,135    
431  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0       265.0      20.0         23.0   
1    $21,761          2.0        4.0       200.0      27.0         31.0   
2    $24,647          2.4        4.0       200.0      25.0         29.0   
3    $30,299          3.2        6.0       270.0      23.0         28.0   
4    $39,014          3.5        6.0       225.0      21.0         24.0   
..        ...         ...        ...         ...       ...          ...   
427  $38,203          2.4        5.0       197.0      24.0         28.0   
428  $40,083          2.3        5.0       242.0      23.0         26.0   
429  $42,573          2.9        6.0       268.0      22.0         26.0   
430  $24,641          1.9        4.0       170.0      25.0         29.0   
431  $33,112          2.5        5.0       208.0      23.0         27.0   

     Weight  Wheelbase  Length  
0    4451.0      106.0   189.0  
1    2778.0      101.0   172.0  
2    3230.0      105.0   183.0  
3    3575.0      108.0   186.0  
4    3880.0      115.0   197.0  
..      ...        ...     ...  
427  3450.0      105.0   186.0  
428  3450.0      105.0   186.0  
429  3653.0      110.0   190.0  
430  2822.0      101.0   180.0  
431  3823.0      109.0   186.0  

[432 rows x 15 columns]